packages/elements/src/datetime-picker/index.ts

```diff
@@ -535,6 +536,35 @@ export class DatetimePicker extends FormFieldElement implements MultiValue {
 
     void super.performUpdate();
   }
+  /**
+   * if the time-picker input(s) is invalid
+   * it will revert time-picker value to previous valid value that store in datetime-picker
+   * @param opened True if opened
+   * @returns {void}
+   */
+  private validateTimePickerInput(): void {
+    if (!this.timepicker && !this.opened) {
+      return;
+    }
+
+    const validateAndFallback = (element: TimePicker | null | undefined, value: string) => {
+      if (!element) {
+        return;
+      }
+
+      const validity = element.checkValidity();
+      if (!validity || (validity && !element.value && value)) {
```

Comment:
* If `!validity` evaluates to `false`, `validity` must be `true`. Therefore, `validity` in `(validity && !element.value && value)` is redundant. 
* Just call `checkValidity()` directly as it would do so only once.

Hope you could get it right this time.

Reply:
If we don't add `(validity && !element.value && value)` condition, it will not sync value back to time-picker when all segments are empty.

Reply:
Just call `checkValidity()` directly.

Reply:
Done